### PR TITLE
Add upgrade-config setting restoreVM document

### PR DESCRIPTION
**Pull request by brandboat**
related to https://github.com/harvester/harvester/issues/4005, add doc for new upgrade-config setting, `restoreVM`.

### CODE REVIEW

File: docs/advanced/settings.md [hunk 1]
```diff
@@ -546,6 +546,8 @@ https://your.upgrade.checker-url/v99/checkupgrade
 
   :::
 
+- `restoreVM`: This setting controls whether to automatically restore running VMs after a single-node upgrade. By default, it is set to `false`, meaning all VMs are stopped after the upgrade. If set to true, Harvester will restart VMs that were running prior to the upgrade. Note that VMs that were paused before the upgrade will remain stopped once the upgrade is complete.
```
Comment (jillian-maroket) [inline docs/advanced/settings.md:549]:
```suggestion
- `restoreVM`: Option that enables Harvester to automatically restore running VMs after a single-node cluster is upgraded. The default value is `false`, which causes all VMs to be stopped after the upgrade is completed. When set to `true`, Harvester restarts VMs that were running before the upgrade was started. VMs that were paused before the upgrade are not restarted.
```

File: docs/advanced/singlenodeclusters.md [hunk 2]
```diff
@@ -55,6 +55,6 @@ If you want Longhorn to create multiple replicas on a node with multiple disks,
 
 ## Upgrades and Maintenance
 
-Single-node clusters do not support [Live Migration](../vm/live-migration.md), so VMs become unavailable during cluster upgrades. Harvester forcibly shuts down all VMs before starting the upgrade process.
+Single-node clusters do not support [Live Migration](../vm/live-migration.md), so VMs become unavailable during cluster upgrades. Harvester forcibly shuts down all VMs before starting the upgrade process. To restore running VMs after upgrade, please set `restoreVM` to true in the [`upgrade-config` setting](./settings.md#upgrade-config).
```
Comment (jillian-maroket) [inline docs/advanced/singlenodeclusters.md:58]:
```suggestion
Single-node clusters do not support [Live Migration](../vm/live-migration.md), so VMs become unavailable during cluster upgrades. Harvester forcibly shuts down all VMs before starting the upgrade process. You can configure Harvester to automatically restore running VMs using the `restoreVM` option of the [`upgrade-config`](./settings.md#upgrade-config) setting.
```

**Review by jillian-maroket — COMMENTED** [review]
Let me know if you have concerns about the changes.

**Review by starbops — APPROVED** [review]
LGTM in the tech aspect. Thank you.

File: docs/advanced/singlenodeclusters.md (hunk 2 above)
Comment (brandboat) [reply docs/advanced/singlenodeclusters.md]: Thanks for reviewing, all comments are addressed.

**Review by jillian-maroket — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**